### PR TITLE
VolumeGet implementation for VMware FCD

### PR DESCRIPTION
**Pull request by hakanmemisoglu**
## Change Overview

VolumeGet implementation for FCD.

## Pull request type

Please check the type of change your PR introduces:
- [ ] Work in Progress
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Trivial/Minor
- [ ] Bugfix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

File: pkg/blockstorage/vmware/vmware.go [hunk 1]
```diff
@@ -72,7 +72,18 @@ func (p *fcdProvider) VolumeDelete(ctx context.Context, volume *blockstorage.Vol
...
-	return nil, errors.New("Not implemented")
+	obj, err := p.gom.Retrieve(ctx, vimID(id))
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to query the disk")
+	}
+	kvs, err := p.gom.RetrieveMetadata(ctx, vimID(id), nil, "")
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to get volume metadata")
+	}
+	tags := convertKeyValueToTags(kvs)
```
Comment (tdmanv) [inline pkg/blockstorage/vmware/vmware.go]: you could also have this return kv directly.

File: pkg/blockstorage/vmware/conversion.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+	"github.com/kanisterio/kanister/pkg/blockstorage"
+)
+
+func convertFromObjectToVolume(vso *types.VStorageObject) *blockstorage.Volume {
+	return &blockstorage.Volume{
+		Type:         blockstorage.TypeFCD,
+		ID:           vso.Config.Id.Id,
+		CreationTime: blockstorage.TimeStamp(vso.Config.CreateTime),
+		Size:         vso.Config.CapacityInMB / 1024,
```
Comment (tdmanv) [inline pkg/blockstorage/vmware/conversion.go]: do we need any nil checks? will config be non nil?

**Comment by tdmanv** [comment]
I added the DNM tag incase you wanted to address the feedback first

File: pkg/blockstorage/vmware/conversion.go (hunk 2 above)
Comment (hakanmemisoglu) [reply pkg/blockstorage/vmware/conversion.go]: They are not pointers, they must be safe.

**Pull request merged**